### PR TITLE
Git hooks v2

### LINKED ISSUE #97: Create git hook to run npm run check before commit
[Project/Issue Tracker](https://github.com/Code-4-Community/lucys-love-bus-frontend/projects/1)

### Objective

I hate having things break in master 😭 I would like to create a git hook that just runs npm run check every time i commit so that I never forget to run it

### PR DESCRIPTION
**Pull request by rymaju**
## Issue

Closes #97 

## Changes

After some thinking I realized not being able to commit anything if any types fail or eslint fails kind of sucks, especially since we're checking for that in CI anyways. I just limited it to prettier-fix instead.

## How this works
After you run `npm install` youre going to install `husky`🐺 which is a package that will run an arbitrary shell script on git hooks. Before you commit anything prettier-fix is going to run. Note that it wont be included in the commit that you just sent, but it will appear in your unstaged changes.

### CODE REVIEW

**Comment by rymaju** [comment]
PR #100 🥳

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+# runs prettier fix before every commit
+npm run prettier-fix
```
Comment (anirudh723) [inline .husky/pre-commit:5]: Add a newline

File: package.json [hunk 2]
```diff
@@ -49,7 +49,8 @@
     "prettier": "prettier --check src",
     "prettier-fix": "prettier --write src",
     "type-check": "tsc",
-    "check": "npm run lint-fix && npm run prettier-fix && npm run type-check"
+    "check": "npm run lint-fix && npm run prettier-fix && npm run type-check",
+    "prepare": "husky install"
```
Comment (anirudh723) [inline package.json:53]: 🐕

**Review by anirudh723 — APPROVED** [review]
Thanks for adding this. As long as you've tested it, ready to 🚢

**Pull request merged**